app/src/main/java/com/babylon/wallet/android/data/repository/cache/database/locker/AccountLockerVaultItemEntity.kt

```diff
@@ -0,0 +1,93 @@
+package com.babylon.wallet.android.data.repository.cache.database.locker
+
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import com.babylon.wallet.android.data.gateway.generated.models.AccountLockerVaultCollectionItem
+import com.babylon.wallet.android.data.gateway.generated.models.AccountLockerVaultCollectionItemFungible
+import com.babylon.wallet.android.data.gateway.generated.models.AccountLockerVaultCollectionItemNonFungible
+import com.babylon.wallet.android.data.gateway.generated.models.AccountLockerVaultCollectionItemType
+import com.radixdlt.sargon.AccountAddress
+import com.radixdlt.sargon.AccountLockerClaimableResource
+import com.radixdlt.sargon.Decimal192
+import com.radixdlt.sargon.LockerAddress
+import com.radixdlt.sargon.ResourceAddress
+import com.radixdlt.sargon.VaultAddress
+import com.radixdlt.sargon.extensions.SargonException
+import com.radixdlt.sargon.extensions.compareTo
+import com.radixdlt.sargon.extensions.init
+import com.radixdlt.sargon.extensions.toDecimal192
+
+@Entity(primaryKeys = ["locker_address", "account_address", "resource_address", "amount", "total_count"])
```

Comment:
For a combination of account, locker and a resource, will there might be additional entries? If yes, does `amount` and `total_count` suffice ?

Reply:
Also maybe `isFungible` is not necessary. Since `amount` is for fungibles and `total_count` is for nfts I suppose. You can distinguish it by those right?

Reply:
I tend to write the code defensively. Having multiple entries stored would have covered all the possibilities. But I've just tested multiple airdrops for the same account/locker/resource combination and having multiple entries is unnecessary, the `amount` and `total_count` are updated and properly returned by the GW. I may have misunderstood the first comment, but my conclusion now is that I have to remove the `amount` and `total_count`  from the primary key definition.

Reply:
I'd rather keep `isFungible` field because it improves code readability by indicating that there are 2 types of entities just by looking at the entity class. Please let me know if there are better ways (and as simple as this?) of modeling this entity though.

Reply:
> I tend to write the code defensively. Having multiple entries stored would have covered all the possibilities. But I've just tested multiple airdrops for the same account/locker/resource combination and having multiple entries is unnecessary, the amount and total_count are updated and properly returned by the GW. I may have misunderstood the first comment, but my conclusion now is that I have to remove the amount and total_count from the primary key definition.

Thanks for looking into this, yes this is what I meant.

> I'd rather keep isFungible field because it improves code readability by indicating that there are 2 types of entities just by looking at the entity class. Please let me know if there are better ways (and as simple as this?) of modeling this entity though.

I am against your solution. Either solution has its drawbacks. What I had done in other entity at some time was to provide a simple getter for such boolean value that would derive its value from the existence of `amount` or `total_count`. Meaning that the `isFungible` can become a backing property (I think that it needs to be ignored from room's serialization) but the drawback is that both `amount` and `total_count` must become nullable. 
So yeah either solution will suffice